compiler/rustc_middle/src/mir/query.rs

```diff
@@ -338,20 +338,17 @@ pub struct ClosureOutlivesRequirement<'tcx> {
     pub blame_span: Span,
 
     // ... due to this reason.
-    pub category: ConstraintCategory,
+    pub category: ConstraintCategory<'tcx>,
 }
 
-// Make sure this enum doesn't unintentionally grow
-rustc_data_structures::static_assert_size!(ConstraintCategory, 12);
-
```

Comment:
Why remove the assert? (Are there problems when there is a lifetime?)

Reply:
Because `SubstsRef` doesn't have a static size I think, at least I was hitting that assertion.

Reply:
Eh that's nonsense, of course it must have a static size. I really don't know why I was hitting those assertions, but the first two CI runs failed because of that, didn't have that problem locally.